dvc/repo/gc.py

```diff
@@ -85,6 +94,11 @@ def gc(  # noqa: PLR0913
             ).values():
                 used_obj_ids.update(obj_ids)
 
+    if not_in_remote:
+        cloud_odb = self.cloud.get_remote_odb(remote, "gc -c")
```

Comment:
Generally, I think we want to push users away from specifying remotes as CLI flags and towards using the `remote:` field in `dvc.yaml`. If I specify different remotes there, will `gc` check those remotes for the appropriate outputs?

Reply:
> f I specify different remotes there, will gc check those remotes for the appropriate outputs?

I need to actually test. I think it would not work as expected in the current P.R.
I also think that it might never worked for `dvc gc -c`

Reply:
I think it's better to codify where the data is stored than have to remember which remote was used. I think there are some existing discussions about it, but I need to find them.

Reply:
>  I also think that it might never worked for dvc gc -c

I have confirmed that `remote:` field in the `.dvc` files doesn't work (never has) with `--cloud`. 

@dberenbaum Do we want to expand the scope of the issue to respect `remote:` field in both the new `--not-in-remote` and the existing `--cloud`?

Reply:
It doesn't have to block this PR, but I think we should still follow up to address it since it's essentially a bug in gc.

Reply:
Updated the P.R. to handle `remote` both in `--cloud` and `--not-in-remote`.